front-end/react-crm/src/pages/auth/Login.tsx

```diff
@@ -20,6 +20,9 @@ declare global {
 export default function Login() {
     const navigate = useNavigate()
     const [token, setToken] = useState(false)
+    const [name, setName] = useState(''); // State for user name
```

Comment:
Hani, this is the file for Google auth login. I have used this file to do my task. Your logic seems like it is correct. But you need to create a new component for your logic. (for example named "customAuth")